File: scripts/_mysql.php
<?php

// -------------------------------------------------------------
// MySQL Services Configuration
// -------------------------------------------------------------

if (empty($config['mysql'])) {
    return;
}


$pma_hosts = [];

foreach ($config['mysql'] as $mysql) {
    $version = $mysql['version'];
    $port = $mysql['port'];
    $host = sprintf('mysql_%s', str_replace('.', '', $version));

    $compose['services'][$host] = [
        'build' => [
            'context' => 'docker/mysql',
            'args' => ['VERSION' => $version],
        ],
        'volumes' => [
            sprintf('./var/mysql/%s:/var/lib/mysql', $version),
        ],
        'ports' => [
            sprintf('%d:3306', $port),
        ],
    ];

    $compose['services']['phpmyadmin']['links'][] = $host;
    $pma_hosts[] = $host;
}

$compose['services']['phpmyadmin']['environment']['PMA_HOSTS'] = implode(',', $pma_hosts);

File: generator/mysql.py
def add_configuration(compose, config):
    mysql_hosts = []
    for version in config['versions']:
        port = config['versions'][version]
        host = 'mysql_'+version.replace('.', '')
        mysql_hosts.append(host)

        compose['services'][host] = {
            'build': {
                'context': 'docker/mysql',
                'args': { 'VERSION': version }
            },
            'volumes': [
                './var/mysql/%s:/var/lib/mysql' % (version)
            ],
            'ports': [
                '%d:3306' % (port)
            ],
        }

    compose['services']['phpmyadmin']['links'] = mysql_hosts
    compose['services']['phpmyadmin']['environment']['PMA_HOSTS'] = ','.join(mysql_hosts)
